core/00_primordials.js

```diff
@@ -436,6 +442,24 @@
   primordials.PromisePrototypeCatch = (thisPromise, onRejected) =>
     PromisePrototypeThen(thisPromise, undefined, onRejected);
 
+  primordials.SafePromiseAll = (iterable) =>
+    // Wrapping on a new Promise is necessary to not expose the SafePromise
+    // prototype to user-land.
```

Comment:
Is it unsafe to expose `SafePromise.prototype` to user-land? I think it can't be modified even if it's exposed because it's frozen. (I might be missing something...

Reply:
This implementation is referenced from `SafePromisePrototypeFinally`. I think it is meant to prevent the internal code from being strangely utilized in user-land.
https://github.com/denoland/deno/blob/45ac6e602da9cd016d4a3b093fd4873b90897f9a/core/00_primordials.js#L439-L455